datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -288,6 +308,18 @@ select unnest(array_remove(column1, 12)) from unnest_table;
 5
 6
 
+## unnest struct-typed column and list-typed column at the same time
+query I?II?
+select unnest(column1), column1, unnest(column5), column5 from unnest_table;
+----
+1 [1, 2, 3] 1 2 {c0: 1, c1: 2}
+2 [1, 2, 3] 1 2 {c0: 1, c1: 2}
+3 [1, 2, 3] 1 2 {c0: 1, c1: 2}
+4 [4, 5] 3 4 {c0: 3, c1: 4}
+5 [4, 5] 3 4 {c0: 3, c1: 4}
+6 [6] NULL NULL NULL
+12 [12] 7 8 {c0: 7, c1: 8}
+
```

Comment:
Can we please add some other tests:

1. Test the output type (e.g. `select arrow_typeof(unnest(column1))`)
2. Test nested structs (I tested it works well, but I think we should have some coverage)


Nested structs
```sql
> create or replace table t as values (struct('a', 'b', struct('c'))), (struct('d', 'e', struct('f')));
0 row(s) fetched.
Elapsed 0.031 seconds.

> select * from t;
+-----------------------------+
| column1                     |
+-----------------------------+
| {c0: a, c1: b, c2: {c0: c}} |
| {c0: d, c1: e, c2: {c0: f}} |
+-----------------------------+
2 row(s) fetched.
Elapsed 0.006 seconds.

> select unnest(column1) from t;
+----------------------+----------------------+----------------------+
| unnest(t.column1).c0 | unnest(t.column1).c1 | unnest(t.column1).c2 |
+----------------------+----------------------+----------------------+
| a                    | b                    | {c0: c}              |
| d                    | e                    | {c0: f}              |
+----------------------+----------------------+----------------------+
2 row(s) fetched.
Elapsed 0.013 seconds.
```

And 

```sql
> create or replace table t as values (struct('a', 'b', [1,2,3])), (struct('x', 'y', [10,20]));
0 row(s) fetched.
Elapsed 0.010 seconds.

> select unnest(column1) from t;
+----------------------+----------------------+----------------------+
| unnest(t.column1).c0 | unnest(t.column1).c1 | unnest(t.column1).c2 |
+----------------------+----------------------+----------------------+
| a                    | b                    | [1, 2, 3]            |
| x                    | y                    | [10, 20]             |
+----------------------+----------------------+----------------------+
2 row(s) fetched.
Elapsed 0.008 seconds.
```

Reply:
Interestingly the output type doesn't seem to work (which probably makes sense as unnest needs to be converted to a LogicalPlan). I don't think we have to fix this in the PR (but adding coverage would be good)

```sql
> select unnest(column1) from t;
+----------------------+----------------------+----------------------+
| unnest(t.column1).c0 | unnest(t.column1).c1 | unnest(t.column1).c2 |
+----------------------+----------------------+----------------------+
| a                    | b                    | [1, 2, 3]            |
| x                    | y                    | [10, 20]             |
+----------------------+----------------------+----------------------+
2 row(s) fetched.

> select arrow_typeof(unnest(column1)) from t;
Internal error: unnest on struct can ony be applied at the root level of select expression.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

Reply:
thank you, i added more coverage